src/executor/evaluate/mod.rs

```diff
@@ -197,6 +197,88 @@ pub async fn evaluate<'a, T: 'static + Debug>(
         Expr::Wildcard | Expr::QualifiedWildcard(_) => {
             Err(EvaluateError::UnreachableWildcardExpr.into())
         }
+        Expr::Case {
+            operand,
+            when_then,
+            else_result,
+        } => {
+            use super::EvaluateError;
+            use itertools::Itertools;
+            use std::mem::discriminant as disc;
+
+            let operand = match operand {
+                Some(expr) => Some(eval(expr).await?),
+                None => None,
+            };
+
+            let when_then = stream::iter(when_then.iter())
+                .map(Ok::<_, Error>)
+                .and_then(|(when, then)| async move { Ok((eval(when).await?, eval(then).await?)) })
+                .try_collect::<Vec<_>>()
+                .await?;
+
+            let else_result = match else_result {
+                Some(result) => Some(eval(result).await?),
+                None => None,
+            };
+
+            let (_, thens) = when_then
+                .clone()
+                .into_iter()
+                .unzip::<_, _, Vec<_>, Vec<_>>();
+
+            let results = match &else_result {
+                Some(er) => [thens, [er.to_owned()].to_vec()].concat(),
+                None => thens,
+            };
+
+            if !results
+                .into_iter()
+                .filter_map(|result| result.try_into().ok())
```

Comment:
You should not ignore `Err`, this is even not an unreachable case.

Reply:
Aha, that's a good point. I'll fix it.